cms/djangoapps/contentstore/utils.py

```diff
@@ -396,16 +396,34 @@ def get_settings_url_for_category(cls, category, child, parent):
             - Problems: Unit url
         """
         handler = category
+        url = '//' + settings.CMS_BASE
+
+        if category == 'chapter':
+            url += '/course/'
+            url += unicode(parent.id)
+            url += '?show='
+            url += unicode(child.location)
+            return url
+
+        elif category == 'sequential':
+            url += '/course/'
+            url += unicode(child.location.course_key)
+            url += '?show='
+            url += unicode(child.location)
+            return url
+
+        elif category == 'vertical':
+            url += '/container/'
+            url += unicode(child.location)
+            return url
 
-        if category == "chapter":
-            usage_key_string = unicode(parent.id)
-        elif category == "sequential" or category == "unit":
-            usage_key_string = unicode(child.location)
         else:
-            usage_key_string = unicode(parent.location)
-            handler = 'unit'
-
-        return reverse('component_handler', kwargs={'usage_key_string': usage_key_string, 'handler': handler})
+            assertEqual(handler, 'problem')
```

Comment:
Okay!  Will remove assertEqual and 'handler' variable, raise an exception to handle edge case, and refactor code so the 'url+=' is not as repetitive.  Will ask later about designing good unit tests.